File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";
        
        string prefix = strs[0];
        
        for (int i = 1; i < strs.size(); i++) {
            int j = 0;
            // Compare characters until the end of current string or prefix
            while (j < strs[i].size() && j < prefix.size() && strs[i][j] == prefix[j]) {
                j++;
            }
            prefix = prefix.substr(0, j);
            
            // Early exit if prefix becomes empty
            if (prefix.empty()) break;
        }
        
        return prefix;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
     public String longestCommonPrefix(String[] strs) 
     {
        if (strs.length == 0) 
            return "";
        String prefix = strs[0]; // flower
        for (int i = 1; i < strs.length; i++)
        {
            // While the current string does not start with the prefix
            while (strs[i].indexOf(prefix) != 0) {
                // Shorten the prefix by removing the last character
                prefix = prefix.substring(0, prefix.length() - 1);
                // If the prefix becomes empty, return an empty string
                if (prefix.isEmpty()) 
                    return "";
            }
        }
        return prefix;
    }
}